src/quo2/components/avatars/channel_avatar/view.cljs

```diff
@@ -0,0 +1,63 @@
+(ns quo2.components.avatars.channel-avatar.view
+  (:require [clojure.string :as string]
+            [quo2.components.avatars.channel-avatar.style :as style]
+            [quo2.components.icon :as icons]
+            [quo2.components.markdown.text :as text]
+            [quo2.foundations.colors :as colors]
+            [react-native.core :as rn]
+            utils.string))
+
+(defn- initials
+  [full-name amount-initials color]
+  [text/text
+   {:accessibility-label :initials
+    :size                :paragraph-2
+    :number-of-lines     1
+    :ellipsize-mode      :clip
+    :weight              :semi-bold
+    :style               {:color color}}
+   (utils.string/get-initials full-name amount-initials)])
+
+(defn- lock
+  [locked? big?]
+  ;; When `locked?` is nil, we must not display the unlocked icon.
+  (when (boolean? locked?)
```

Comment:
I completely agree @cammellos, but this would be a refactor since the original code used a boolean to represent 3 states. I felt the PR was growing more than I wanted.